engine/crates/composition/src/subgraphs/definitions.rs

```diff
@@ -90,3 +90,17 @@ impl<'a> DefinitionWalker<'a> {
         self.walk(self.definition().subgraph_id)
     }
 }
+
+impl<'a> SubgraphWalker<'a> {
+    pub(crate) fn definitions(self) -> impl Iterator<Item = DefinitionWalker<'a>> {
+        let subgraph_id = self.id;
+        let definitions = &self.subgraphs.definitions.0;
+        let start = definitions.partition_point(|def| def.subgraph_id < self.id);
+        let subgraph_definitions = definitions[start..]
```

Comment:
This stuff smells like something we could have in a more generic way available somewhere.

Reply:
See `RangeWalker` in @Finistere 's introspection PR.